modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java

```diff
@@ -89,86 +157,468 @@ public IgniteStatisticsRepository statisticsRepository() {
     }
 
     /** {@inheritDoc} */
-    @Override public void clearObjectStatistics(String schemaName, String objName, String... colNames) {
-        StatsKey key = new StatsKey(schemaName, objName);
+    @Override public void clearObjectStatistics(
+            String schemaName,
+            String objName,
+            String... colNames
+    ) throws IgniteCheckedException {
+        final StatsCollectionFutureAdapter doneFut = new StatsCollectionFutureAdapter();
+
+        UUID reqId = UUID.randomUUID();
+        StatsKeyMessage keyMsg = StatisticsUtils.toMessage(schemaName, objName, colNames);
+
+        Map<UUID, List<Integer>> requestNodes = null;
+        /*try {
+            requestNodes = nodePartitions(extractGroups(Collections.singletonList(keyMsg)), null);
+        } catch (IgniteCheckedException e) {
+            // TODO: handle & remove task
+        }
+
+        StatsClearRequest req = new StatsClearRequest(reqId, false, Collections.singletonList(keyMsg));
+
+        sendLocalRequests(reqId, req, requestNodes);*/
+
+        clearObjectStatisticsLocal(keyMsg);
+    }
+
+    /**
+     * Actually clear local object statistics by the given key.
+     *
+     * @param keyMsg Key to clear statistics by.
+     */
+    private void clearObjectStatisticsLocal(StatsKeyMessage keyMsg) {
+        StatsKey key = new StatsKey(keyMsg.schema(), keyMsg.obj());
+        String[] colNames = keyMsg.colNames().toArray(new String[0]);
```

Comment:
Why is colNames a part of StatsKeyMessage, but not a part of StatsKey?

Reply:
Because we store statistics by object, aquire statistics by whole object, but we can send request to collect statistics by only some of the tables columns.